base/applications/network/dwnl/dwnl.c

```diff
@@ -246,6 +247,10 @@ CBindStatusCallback_OnProgress(IBindStatusCallback *iface,
             break;
 
         case BINDSTATUS_ENDDOWNLOADDATA:
+            /* Since download is completed, update progress one last time to be at 100% */
+            This->Progress = This->Size; // Ensure progress == total size
```

Comment:
OK I won't yet merge the PR today because I have a dumb question: What happens if the download unexpectedly stops (like for example you cut off the internet connection in the middle of the download)? The download should "end" I suppose, but be left incomplete, no? Is `BINDSTATUS_ENDDOWNLOADDATA` called even in this scenario? If so, it shouldn't show 100% progress. Or is this scenario handled in another place in dwnl?

Reply:
Hmm. Let me try to address this

Reply:
Thanks for your detailed analysis @TAbdiukov ; if there's something to add, it would be better to do it in a separate PR. I'll merge this one.